Cosmos type-check failing with latest release of `apache-airflow-providers-cncf-kubernetes`

There was a [release](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/changelog.html#changelog) today that [added type hints](https://github.com/apache/airflow/pull/36048/files) to the provider function `convert_env_vars` which Cosmos uses below in the `KubernetesBaseOperator`:

https://github.com/astronomer/astronomer-cosmos/blob/e4278c625f971923a2d0f4779edf9592a2314972/cosmos/operators/kubernetes.py#L56-L62

Currently `env` has type hints of `dict[str, str | bytes | PathLike[Any]]` which is not compatible and is resulting in the type check error below:

```shell
cosmos/operators/kubernetes.py:62: error: Unpacked dict entry 0 has incompatible type "dict[str, Union[str, bytes, PathLike[Any]]]"; expected "SupportsKeysAndGetItem[str, str]"  [dict-item]
```

--- a/cosmos/operators/kubernetes.py
+++ b/cosmos/operators/kubernetes.py
@@ -56,10 +56,12 @@ class DbtKubernetesBaseOperator(KubernetesPodOperator, DbtBaseOperator):  # type
     def build_env_args(self, env: dict[str, str | bytes | PathLike[Any]]) -> None:
         env_vars_dict: dict[str, str] = dict()
 
+        for env_var_key, env_var_value in env.items():
+            env_vars_dict[env_var_key] = str(env_var_value)
         for env_var in self.env_vars:
             env_vars_dict[env_var.name] = env_var.value
 
-        self.env_vars: list[Any] = convert_env_vars({**env, **env_vars_dict})
+        self.env_vars: list[Any] = convert_env_vars(env_vars_dict)
 
     def build_and_run_cmd(self, context: Context, cmd_flags: list[str] | None = None) -> Any:
         self.build_kube_args(context, cmd_flags)

